lib/src/scene.dart

```diff
@@ -4,12 +4,30 @@ import 'environment/state/environment_state.dart';
 
 /// Signature for a function that creates a widget and provides a hook to
 /// trigger a rebuild of the current [Scene].
-typedef EnvironmentControlBuilder = Widget Function(
+typedef EnvironmentControlBuilder<T> = Widget Function(
   BuildContext context,
+  T currentValue,
   EnvironmentState environmentState,
-  // VoidCallback rebuildScene,
 );
 
+class EnvironmentControl<T> {
+  EnvironmentControl({
```

Comment:
Is there a reason why this object can't just be owned by the Scene and take a persistent reference to EnvironmentState? That way, it could expose something like a `currentValue` property, and the scene builder could reference the typesafe value directly from the relevant EnvironmentControl.

Reply:
> take a persistent reference to EnvironmentState

I don't have any objection to this conceptually, but I think it would require a pretty big change to implement. Currently: 
- `StagerApp` owns the EnvironmentState and accepts already-constructed Scene instances. We would need to move the creation of EnvironmentState outside of `StagerApp` and pass it to both `StagerApp` and all Scenes.
- EnvironmentControls don't have a notion of the Scene they're attached to (and a single control could be shared by multiple scenes, I imagine this will be a reasonably common use case), so we would need a mechanism to provide EnvironmentState independent of a particular Scene.

`currentValue` would be really easy to add, though, and I could see that being useful.